### PR TITLE
Add alert if backup sidecar metrics are not scrapped

### PR DESCRIPTION
**Pull request by swapnilgm**
Signed-off-by: Swapnil Mhamane <swapnil.mhamane@sap.com>

**What this PR does / why we need it**:
In case the backup sidecar is configured wrongly, or metrics data points are not update by backup sidecar due to some reason. Then it could mean that backups are not being processed.
Alert-manager with current rule don't raise alerts in above case.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
cc @wyb1 agreed to add alerting rule test for this new alert.
@shreyas-s-rao check if alerting expression in fine from etcd-backup-restore point of view.
 
**Release note**:

```improvement operator
None
```

### CODE REVIEW

**Comment by rfranzke** [comment]
Ideally, all of the etcd specific alerts should be moved to etcd-druid and out of g/g. There is a contract how to contribute alerts/dashboards/scrape config: https://github.com/gardener/gardener/blob/master/docs/extensions/logging-and-monitoring.md

WDYT?

File: charts/seed-monitoring/charts/core/charts/prometheus/rules/kube-etcd3.rules.yaml
Comment (shreyas-s-rao) [inline charts/seed-monitoring/charts/core/charts/prometheus/rules/kube-etcd3.rules.yaml]:
```suggestion
    expr: (absent(etcdbr_snapshot_required{job="kube-etcd3-backup-restore",role="main"}) == bool 1) + (changes(etcd_debugging_mvcc_current_revision{job="kube-etcd3",role="main"}) >= bool 1) == 2
```

**Review by shreyas-s-rao — CHANGES_REQUESTED** [review]
@swapnilgm  Thanks for the PR. The current alert expression works for a general shoot, wouldn't work in the case where there are no shoot operations, because of which there are no updates to etcd. You would have to use the `etcd_debugging_mvcc_current_revision` to check if there's been an etcd update yet the `etcdbr_snapshot_required` metric is absent. Also add `etcd_debugging_mvcc_current_revision` to the list of allowed KubeETCD3 metrics in prometheus `values.yaml`.

**Comment by swapnilgm** [comment]
> Ideally, all of the etcd specific alerts should be moved to etcd-druid and out of g/g. There is a contract how to contribute alerts/dashboards/scrape config: https://github.com/gardener/gardener/blob/master/docs/extensions/logging-and-monitoring.md
> 
> WDYT?

Makes sense to me. But lets just go ahead with this PR. And take this as separate task.

**Review by rfranzke — APPROVED** [review]
Sure. @shreyas-s-rao, please review/approve.

**Comment by shreyas-s-rao** [comment]
/lgtm from my side
Waiting for @wyb1 's review before going ahead and merging

**Pull request merged**
